spec/requests/api/v1/targets/destroy_spec.rb

```diff
@@ -0,0 +1,54 @@
+describe 'DELETE api/v1/targets/:id', type: :request do
+  let!(:user)               { create(:user) }
+  let!(:user_targets)       { create_list(:target, 3, user:) }
+  let!(:other_targets)      { create_list(:target, 3) }
+  let(:first_target_user)   { user_targets.first }
+  let(:failed_response)     { 404 }
+  let(:id)                  { first_target_user.id }
+  subject { delete api_v1_target_path(id:), headers: auth_headers, as: :json }
+
+  it 'returns a successful response' do
+    subject
+    expect(response).to be_successful
+  end
+
+  it 'deletes the target' do
+    expect { subject }.to change(Target, :count).from(6).to(5)
+  end
+
+  context 'when target do not belongs to current user' do
+    let(:id) { other_targets.first.id }
+
+    it 'does not delete a target' do
+      expect { subject }.not_to change { Target.count }
+    end
+
+    it 'does not return a successful response' do
+      subject
+      expect(response.status).to eq(failed_response)
+    end
+  end
+
+  context 'when the id is not valid' do
```

Comment:
what about adding a context  to try to delete a Target from other user ?

Reply:
Done!